tests/test_dot.py

```diff
@@ -1,52 +1,374 @@
+
+
+# TODO: get rid of falls back on NumPy when tensordot
+# is implemented using OneMKL
+@pytest.mark.usefixtures("allow_fall_back_on_numpy")
+@pytest.mark.parametrize("dtype", get_complex_dtypes())
+@pytest.mark.parametrize(
+    "array_info",
+    [
+        (1, 10, (), (10,)),
+        (10, 1, (10,), ()),
+        (1, 1, (), ()),
+        (10, 10, (10,), (10,)),
+        (12, 6, (4, 3), (3, 2)),
+        (12, 3, (4, 3), (3,)),
+        (60, 3, (5, 4, 3), (3,)),
+        (4, 8, (4,), (4, 2)),
+        (60, 48, (5, 3, 4), (6, 4, 2)),
+    ],
+    ids=[
+        "0d_1d",
+        "1d_0d",
+        "0d_0d",
+        "1d_1d",
+        "2d_2d",
+        "2d_1d",
+        "3d_1d",
+        "1d_2d",
+        "3d_3d",
+    ],
+)
+def test_dot_complex(dtype, array_info):
+    size1, size2, shape1, shape2 = array_info
+    x11 = numpy.random.uniform(-5, 5, size1)
+    x12 = numpy.random.uniform(-5, 5, size1)
+    x21 = numpy.random.uniform(-5, 5, size2)
+    x22 = numpy.random.uniform(-5, 5, size2)
+    a = numpy.array(x11 + 1j * x12, dtype=dtype).reshape(shape1)
+    b = numpy.array(x21 + 1j * x22, dtype=dtype).reshape(shape2)
+    ia = dpnp.array(a)
+    ib = dpnp.array(b)
+
+    result = dpnp.dot(ia, ib)
+    expected = numpy.dot(a, b)
+    assert_dtype_allclose(result, expected)
+
+
+# TODO: get rid of falls back on NumPy when tensordot
+# is implemented using OneMKL
+@pytest.mark.usefixtures("allow_fall_back_on_numpy")
+@pytest.mark.parametrize("dtype", get_all_dtypes())
+@pytest.mark.parametrize(
+    "array_info",
+    [
+        (1, 10, (), (10,)),
+        (10, 1, (10,), ()),
+        (1, 1, (), ()),
+        (10, 10, (10,), (10,)),
+        (12, 6, (4, 3), (3, 2)),
+        (12, 3, (4, 3), (3,)),
+        (60, 3, (5, 4, 3), (3,)),
+        (4, 8, (4,), (4, 2)),
+        (60, 48, (5, 3, 4), (6, 4, 2)),
+    ],
+    ids=[
+        "0d_1d",
+        "1d_0d",
+        "0d_0d",
+        "1d_1d",
+        "2d_2d",
+        "2d_1d",
+        "3d_1d",
+        "1d_2d",
+        "3d_3d",
+    ],
+)
+def test_dot_ndarray(dtype, array_info):
+    size1, size2, shape1, shape2 = array_info
+    a = numpy.array(numpy.random.uniform(-5, 5, size1), dtype=dtype).reshape(
+        shape1
+    )
+    b = numpy.array(numpy.random.uniform(-5, 5, size2), dtype=dtype).reshape(
+        shape2
+    )
+    ia = dpnp.array(a)
+    ib = dpnp.array(b)
+
+    result = ia.dot(ib)
+    expected = a.dot(b)
+    assert_dtype_allclose(result, expected)
+
+
+@pytest.mark.parametrize("dtype", get_all_dtypes())
+def test_dot_strided(dtype):
+    a = numpy.ones(25, dtype=dtype)
+    b = numpy.ones(25, dtype=dtype)
+    ia = dpnp.array(a)
+    ib = dpnp.array(b)
+
+    result = dpnp.dot(ia[::3], ib[::3])
+    expected = numpy.dot(a[::3], b[::3])
+    assert_dtype_allclose(result, expected)
+
+    # TODO: unmute when the problem with negative stride is fixed
+    # result = dpnp.dot(ia, ib[::-1])
+    # expected = numpy.dot(a, b[::-1])
+    # assert_dtype_allclose(result, expected)
```

Comment:
I'd suggest to extend the test scenario, like
```suggestion
    result = dpnp.dot(ia, ib[::-1])
    expected = numpy.dot(a, b[::-1])
    assert_dtype_allclose(result, expected)

    result = dpnp.dot(ia[::-2], ib[::2])
    expected = numpy.dot(a[::-2], b[::2])
    assert_dtype_allclose(result, expected)

    result = dpnp.dot(ia[::-5], ib[::-5])
    expected = numpy.dot(a[::-5], b[::-5])
    assert_dtype_allclose(result, expected)
```